reco_utils/dataset/pandas_df_utils.py

```diff
@@ -100,45 +103,43 @@ def df_to_libffm(df, col_rating=DEFAULT_RATING_COL, filepath=None):
         filepath (str): path to save the converted data.
 
     Return:
-        df_libffm (pd.DataFrame): a Pandas DataFrame in libffm format.
+        pd.DataFrame: data in libffm format.
     """
     df_new = df.copy()
 
-    # Encode field (rating column is not included)
-    field_names = list(df_new.drop(col_rating, axis=1).columns)
-    field_dict = {k: v+1 for v, k in enumerate(field_names)}
-
-    # Encode field-feature
-    field_feature_list = []
-    for col in field_names:
-        if df_new[col].dtype == int:
-            df_new[col] = df_new[col].astype(str)
-        if df_new[col].dtype == object:
-            field_feature_concat = ['{}-{}'.format(col, x) for x in df_new[col].values]
-            field_feature_list.extend(field_feature_concat)
-    field_feature_dict = {k: v+1 for v, k in enumerate(field_feature_list)}
-
-    def _convert(field, feature, field_index_dict, field_feature_index_dict):
-        field_index = field_index_dict[field]
-        field_feature_index = (
-            field_feature_index_dict['{}-{}'.format(field, feature)]
-            if isinstance(feature, str)
-            else None
-        )
+    # Check column types.
+    types = df_new.dtypes
+    if not all([x == object or x == float or x == int for x in types]):
```

Comment:
Tested this and seems does not work with np.int64

Reply:
Did you test it on Windows? It works on my Linux machine. But good to know as something related to dyptes also failed in the notebook, and it must have something to do with the OS. #616

Reply:
Seems in Windows, C long is not 64bit even you use 64bit os, and python int is C long, so this bug happens...  but actually this behavior gives us good insight about potential bug. 

Let's say the dataset is too large so that users want to reduce the size to save some memory, maybe by calling `pd.to_numeric(s, downcast='integer')` and the result is saved as `np.int16` type. Then we'll have bugs wherever we used a simple type comparison. same to float.

Reply:
Good observation and indeed this is something we need to particularly careful about. @gramhagen @roalexan @miguelgfierro FYI